### DISCUSSION TITLE
Bugs and Continued Development of New Features

### DISCUSSION THREAD

**Discussion by Sonwon1** [osmandapp/OsmAnd#18020] [General]
Just my opinion. OsmAnd just started crashing on me today, version 4.5.10, Android 13. It was working fine yesterday. I have not changed anything and decided to edit the profiles today when it started crashing. Multiple crash reports over six have been sent via email.

I know it is nice to have new features and quality of life fixes however there are to many open bugs, over 200. All development should focus on fixing the existing bugs before releasing anything new. I know it is not fun tracking down bugs however this app is not useful if it keeps crashing.

I am leaving soon on a 16 day trip, all routes are loaded and now it starts crashing. :( OsmAnd needs to be rock stable to be useful and I am seriously concerned.

I am backing OsmAnd up at the moment and will try a wipe and restore to see if that helps, if not it looks like I will be using the Garmin for my trip.

**Comment by Sonwon1** [comment]
I opened a bug report. It appears removing the Map Marker widget from the top panel causes the crash and then a crash whenever you try to enter the profile. I still stand by what I wrote above, to many bugs to many problems to many crashes with OsmAnd. I may move to Locus Maps?

[Removing Map Marker Widget from Driving Profile Causes OsmAnd to Crash #18021](https://github.com/osmandapp/OsmAnd/issues/18021)

**Comment by DmitryAlexei** [comment]
Your issue may be related with this one https://github.com/osmandapp/OsmAnd/issues/17290
How many tracks have you got on the map displayed?

**Reply by Sonwon1** [reply]
Zero

To add some more context this happened on a fresh reload of OsmAnd after I removed the Map Marker widget per my bug report.

**Comment by Sonwon1** [comment]
Now, weather is broken again for me on my Samsung Galaxy Active Tab 3, Android 13. It worked while I was on version 4.5.9. Versions 4.5.8 and 4.5.10 did not work. Funny thing is it works on my phone, Android 10, Motorola One Zoom.

**Comment by vshcherb** [comment]
> I am leaving soon on a 16 day trip, all routes are loaded and now it starts crashing. :( OsmAnd needs to be rock stable to be useful and I am seriously concerned.

Latest version of 4.5.10 considered to be stable, there is no known regressions so far comparing to 4.4 version. You mentioned crash with Map Markers currently it's in reproduced stage, so it was not known yet. 

Crashes might be related that routes are now taking more memory and that needs to be fixed soon, if that's your main concern of tracks and memory usage, I would recommend to switch back to 4.4 release

**Comment by Sonwon1** [comment]
> > I am leaving soon on a 16 day trip, all routes are loaded and now it starts crashing. :( OsmAnd needs to be rock stable to be useful and I am seriously concerned.
> 
> Latest version of 4.5.10 considered to be stable, there is no known regressions so far comparing to 4.4 version. You mentioned crash with Map Markers currently it's in reproduced stage, so it was not known yet.
> 
> Crashes might be related that routes are now taking more memory and that needs to be fixed soon, if that's your main concern of tracks and memory usage, I would recommend to switch back to 4.4 release

None of my routes are over 250 kilometers. I believe the documentation says no problems up to 300 kilometers? 

I believe the screen panels are buggy and causing problems, like [crashes from deleting the Map Marker widgets](https://github.com/osmandapp/OsmAnd/issues/18021), [widgets moving from the right panel to the left panel](https://github.com/osmandapp/OsmAnd/discussions/17807), l[eft panel showing while in navigation menu](https://github.com/osmandapp/OsmAnd/issues/17960). I did not search for other problems in Issues:open these are just the ones I have bumped into. Remember there are many users that just move on or give up when finding a problem rather than report issues here.

Yes if that is how it must be then... It is not the best solution for me. Just because 4.5.10 is not causing issues with the test hardware and OS does not mean I am not having issues. It likely just means I have a unique device, Android version and configuration that is untested with OsmAnd 4.5.10. In my opinion, OsmAnd has not been very friendly to Android 13 or the Samsung Galaxy Tab Active 3 or both? I state that because it appears to be stable on Android 10 running on a Motorola One Zoom. I would offer to beta test however I need stable navigation software so that puts me out of being a tester. I suspect there are enough changes between Android 10 and 13 to break things or the Samsung device has some Samsung unique software causing issues?

I would expect there will be more reports as people upgrade their hardware and OS versions. I am not sure I can wait around? I will know more after this holiday an reevaluate where to go.

**Comment by Sonwon1** [comment]
I am back from my 16 day road tour. I took a Garmin Zumo XT with me as a backup nav solution. As it turned out the Garmin was not needed. I had a few problems with OsmAnd hanging which I cannot reproduce. Restarting OsmAnd cleared up the problem. I was using OsmAnd for my routing which worked almost flawlessly. However there was the one time OsmAnd routing tried to route me on a bicycle trail. That caused some confusion on my part as the spot was at a road split which I turned the wrong way and then was off the route. After we went back and looked at the signs we realized it was a bicycle route. This was not intuitive since we were in a foreign country and did not able to speak the language.

I will try to duplicate the routing problem to see if it is reproducible when I have time. If I can reproduce I will open an issue. I also plan to test with bRouter. Maybe it is a map problem?

Best of all was when the road was under construction and we had to ride through sections of 10 cm deep gravel on a road bike. It was squirrelly and close to dropping the bike however we passed through it. The blessing is there was no road work on that day as it was Sunday. Tip you can use the Google Android Translator app and download a language so the translator will work offline. I had to use that on the construction sign to figure out if the road was passable.

Lastly, I avoided triggering the crashing bug that started this thread by not removing the map marker banner across the top of the screen. I understand this is now fixed in release 4.6. I am wondering when version 4.6 gets released to the general public?

**Reply by mikehgentry** [reply]
If you're on Google Play you can opt in to the beta, that should get you 4.6.x. Obviously it's a beta though, so it'll come with its own problems. You can gauge how close they are to release very roughly by looking at the milestones

https://github.com/osmandapp/OsmAnd/milestones

They won't fix *every* bug tagged for a release before releasing it, but they'll fix most of them. So the short answer is it'll be a few months yet I suspect. That also fits with the pattern of previous releases, they seem to release a major new version *roughly* every six months

https://osmand.net/docs/versions/free-versions

**Reply by Sonwon1** [reply]
Thank you for the reply!

Since I need a solid reliable navigation I will stay away from the beta build. It would be nice if we could load the beta and stable at the same time so we could test the new version without losing stable.
